apps/public/src/app/(general)/bijlage/Bijlage.tsx

```diff
@@ -1,39 +1,66 @@
 'use client'
 
 import { Column, Field, Heading, Label, Paragraph, UnorderedList } from '@amsterdam/design-system-react'
+import type { AttachmentOutput } from '@meldingen/api-client'
 import { Grid, SubmitButton } from '@meldingen/ui'
+import { useState } from 'react'
 
 import { BackLink } from '../_components/BackLink'
 
 import { FileUpload } from './_components/FileUpload'
-import { postAttachmentForm } from './actions'
-
-export const Bijlage = () => (
-  <Grid paddingBottom="large" paddingTop="medium">
-    <Grid.Cell span={{ narrow: 4, medium: 6, wide: 6 }} start={{ narrow: 1, medium: 2, wide: 3 }}>
-      <BackLink href="/locatie" className="ams-mb--xs">
-        Vorige vraag
-      </BackLink>
-      <Heading className="ams-mb--sm">Foto’s</Heading>
-      <form action={postAttachmentForm}>
-        <Field>
-          <Label htmlFor="file-upload" optional>
-            Heeft u een foto om toe te voegen?
-          </Label>
-          <Column id="file-upload-description" className="ams-mb--sm">
-            <Paragraph>
-              Voeg een foto toe om de situatie te verduidelijken. Verwijder alle persoonsgegevens van u en derden.
-            </Paragraph>
-            <UnorderedList>
-              <UnorderedList.Item>U kunt maximaal drie bestanden tegelijk toevoegen.</UnorderedList.Item>
-              <UnorderedList.Item>Toegestane bestandtypes: jpg, jpeg en png.</UnorderedList.Item>
-              <UnorderedList.Item>Een bestand mag maximaal 20 MB groot zijn.</UnorderedList.Item>
-            </UnorderedList>
-          </Column>
-          <FileUpload id="file-upload" aria-describedby="file-upload-description" />
-        </Field>
-        <SubmitButton>Volgende vraag</SubmitButton>
-      </form>
-    </Grid.Cell>
-  </Grid>
-)
+import { postAttachmentForm, redirectToNextPage } from './actions'
+
+const isErrorMessage = (obj: unknown): obj is { message: string } => {
+  return typeof obj === 'object' && obj !== null && 'message' in obj && typeof (obj as any).message === 'string'
+}
+
+export const Bijlage = () => {
+  const [uploadedFiles, setUploadedFiles] = useState<AttachmentOutput[]>([])
+  const [errorMessage, setErrorMessage] = useState<string>()
+
+  const handleOnChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (!event.currentTarget.files) return
+
+    const result = await postAttachmentForm(event.currentTarget.files)
```

Comment:
We can use `useActionState` here to simplify this a little bit. That would also mean we don't have to store the error message as client side state.